lib/rubocop/cop/style/redundant_arguments.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require 'parser/current'
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for redundant arguments to methods.
+      # Method names & arguments can be configured like:
+      #
+      # RedundantArguments:
+      #   join: ''
+      #   split: ' '
+      #   foo: 2
+      #
+      # Limitations:
+      # 1. This cop matches for method names only and hence cannot tell apart
+      #    methods with same name in different classes.
+      # 2. This cop is limited to methods with single parameter.
+      # 3. This cop is unsafe if certain kernel global variables are set.
+      #
+      # @example
+      #   # bad
+      #
+      #   array.join('')
+      #   [1, 2, 3].join("")
+      #   string.split(" ")
+      #   "first\nsecond".split(" ")
+      #   A.foo(2)
+      #
+      #   # good
+      #   array.join
+      #   [1, 2, 3].join
+      #   string.split
+      #   "first second".split
+      #   A.foo
+      class RedundantArguments < Cop
+        MSG = 'Argument is redundant.'
```

Comment:
I'd expand the message to mention the argument in question and some text saying that the argument is implied by default, so it's clearer what makes it redundant.